### PR TITLE
IDE relayout

### PR DESCRIPTION
**Pull request by cpirich**
Applab is going to use a different layout such that the page will not scroll vertically. Various panes will be pinned to the top (header), left (app visualization), right (workspace), and bottom (control program execution). This PR is step one in that process. It involves a lot of change for blockly and droplet layout, but the end result for blockly levels should look the same as today.
I've done manual testing, but this is a big change, so we may find some things in UI tests or embed/levelbuilder scenarios.
- Consolidated codeWorkspace and blockly divs to just codeWorkspace
- Renamed blocklyApp div to codeApp
- Removed unused helpArea from templates
- Substantially rewrote onResize() such that all work is now handled by the browser through CSS except for two parts we should try to clean up later: an RTL case for blockly & the resizeHeaders() code
- Renamed blockly-header to workspace-header
- Removed the footer for applab
- Updated (& tested) our RTL handling
- Updated applab' shadow copy of the responsive CSS classes & the code that manipulates them
- Removed 25px of padding between the header and the top of the puzzle levels

### CODE REVIEW

File: apps/style/common.scss [hunk 1]
```diff
@@ -197,6 +216,12 @@ div.droplet-palette-group-header-selected {
   #visualizationColumn.responsive {
     max-width: $width;
   }
+  #codeWorkspace {
```
Comment (bcjordan) [inline apps/style/common.scss]: These two segments could make a good [mixin](http://www.sass-lang.com/guide#topic-6)

File: apps/style/applab/style.scss [hunk 2]
```diff
@@ -40,12 +54,19 @@ div#visualization {
 }
 
 @media screen and (min-width: 1051px) and (max-width: 1100px) {
+  $width: 300px;
   div#visualization.responsive {
-    max-height: 300px;
-    max-width: 300px;
+    max-height: $width;
```
Comment (bcjordan) [inline apps/style/applab/style.scss]: Potential mixin opportunity (with a `$width` parameter)

**Comment by islemaster** [comment]
This seems like good cleanup to me.  Being able to hide the page footer is great.

**Comment by bcjordan** [comment]
LGTM, nice cleanup!

File: apps/style/applab/style.scss [hunk 3]
```diff
@@ -55,12 +76,19 @@ div#visualization {
...
+    max-width: $width;
   }
   div#visualizationColumn.responsive {
-    max-width: 250px;
+    max-width: $width;
+  }
+  div#codeWorkspace {
+    left: $width;
+  }
+  html[dir='rtl'] div#codeWorkspace {
```
Comment (davidsbailey) [inline apps/style/applab/style.scss]: optional CSS trick for next time: change this line to "html[dir='rtl'] & {" and move it into the div#codeWorkspace block above.

**Pull request merged**